server/src-lib/Hasura/Metadata/Class.hs

```diff
@@ -0,0 +1,98 @@
+-- | This module has type class and types which implements the Metadata Storage Abstraction
+module Hasura.Metadata.Class
+  ( MetadataStorageT(..)
+  , runMetadataStorageT
+  , MonadMetadataStorage(..)
+  )
+where
+
+import           Control.Monad.Morph                    (MFunctor)
+
+import           Hasura.Eventing.HTTP
+import           Hasura.Eventing.ScheduledTrigger.Types
+import           Hasura.Prelude
+import           Hasura.RQL.Types
+
+import qualified Hasura.Tracing                         as Tracing
+
+newtype MetadataStorageT m a
+  = MetadataStorageT {unMetadataStorageT :: ExceptT QErr m a}
+  deriving ( Functor, Applicative, Monad
+           , MonadError QErr
+           , MonadTrans
+           , MonadIO
+           , MFunctor
+           , Tracing.HasReporter
+           )
+
+runMetadataStorageT
+  :: MetadataStorageT m a -> m (Either QErr a)
+runMetadataStorageT =
+  runExceptT . unMetadataStorageT
+
+{- Note [Todo: Common interface for eventing sub-system]
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+Scheduled and Table event triggers have their own implementation
+and tables in Postgres. The core logic for both is same. So it
+is necessary to have a unified interface (maybe via a Postgres extension)
+where all event triggers implementation can extend from. By this we can
+fairly reduce the Postgres schema foot print and interactions to the database.
+
+TODO: Reference to open issue or rfc?
+-}
```

Comment:
Could you please expand this note a little bit? I think an important point is being made here, but I find the language a bit too abstract to grasp.